Author: Lucas Nussbaum <lucas@lucas-nussbaum.net>
Description: patch for Debian bug #341256
replace AC_FUNC_SETPGRP with something that works when cross-compiling
--- a/configure.in
+++ b/configure.in
@@ -831,7 +831,25 @@ main()
 fi
 
 AC_FUNC_GETPGRP
-AC_FUNC_SETPGRP  
+
+dnl AC_FUNC_SETPGRP does not work if cross compiling
+dnl Instead, assume we will have a prototype for setpgrp if cross compiling.
+if test "$cross_compiling" = no; then
+  AC_FUNC_SETPGRP
+else
+  AC_CACHE_CHECK([whether setpgrp takes no argument], ac_cv_func_setpgrp_void,
+    [AC_TRY_COMPILE([
+#include <unistd.h>
+], [
+  if (setpgrp(1,1) == -1)
+    exit (0);
+  else
+    exit (1);
+], ac_cv_func_setpgrp_void=no, ac_cv_func_setpgrp_void=yes)])
+if test $ac_cv_func_setpgrp_void = yes; then
+  AC_DEFINE(SETPGRP_VOID, 1)
+fi
+fi
 
 AC_C_BIGENDIAN
 AC_C_CONST
